### PR TITLE
Detail security policy

### PR DESCRIPTION
**Pull request by emmanuelbernard**
This is a proposal to the Quarkus project security policy.

Open questions are:

1. Are we fine with the supported statement and what we commit to
2. Are we fine to delegate to Red Hat's security team
3. Are we fine with offering to report third party libs issues to us as an ecosystem?

### CODE REVIEW

**Comment by emmanuelbernard** [comment]
I had a conversation with the Red Hat security team. I also had to handle a few security bugs (listed as CVEs or not) in my carrier. My feedback is that this is a process you need to be guided to to do it properly. I think delegating to the Red Hat security team is the best for Quarkus.
Red Hat is trusted on both security and open source commitment.

There is an alternative model where we have a Quarkus secalert team that delegates and interacts to the Red Hat team and others as needed. The feedback from the Red Hat security team is that this has created confusions and grey areas in how things should be handled in their past experience. They do recommend one owner.

**Comment by emmanuelbernard** [comment]
Don't merge, yet, the DRAFT flag cannot be set due to a GitHub bug https://github.com/isaacs/github/issues/1547

File: SECURITY.md
Comment (gsmet) [inline SECURITY.md:24]: `available Red Hat's` -> word missing

File: SECURITY.md
Comment (gsmet) [inline SECURITY.md:35]:
```suggestion
If you find a security bug possibly rooted in one of these libraries, you can either disclose to them directly or disclose them to the Quarkus team (following this process) and we will responsibly disclose the issue to the respective extension or library maintainer.
```

**Comment by stianst** [comment]
@emmanuelbernard For Keycloak we have a separate google groups where community can report issues to the Keycloak team. We have someone from Red Hat security team on the mailing list as well. Not sure what is the ideal, but personally I like to be included in the report, not just having it go to Red Hat security, as they may not prioritise the community-only issues at the same level as we do.

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted some small issues.

**Comment by sberyozkin** [comment]
Someone from the Quarkus team should be indeed updated of the issues immediately IMHO. If we find out about some vulnerability a week or so later after it was reported then it would not be ideal - we can make some code decisions since the moment it was reported which can make it even worse or more difficult to reproduce.

File: SECURITY.md
Comment (mkouba) [inline SECURITY.md:45]: The Red Hat security team is a trusted member ~if~ **of** the security community.

**Comment by emmanuelbernard** [comment]
@stianst When I discussed with the security team, they said the model your describe had the downsides I mentioned. I can come back tot hem with your example. How many people do you have in this security group? Security is need to know basis so I'm not concern to not know. Anyways the Red Hat sec team will always loop a Quarkus person.

File: SECURITY.md
Comment (emmanuelbernard) [reply SECURITY.md:24]: fixed.

File: SECURITY.md
Comment (emmanuelbernard) [reply SECURITY.md:45]: Fixed

**Comment by sberyozkin** [comment]
@emmanuelbernard If we only have a redirector which will forward to secalert@redhat.com and a few more people from the team as you choose then it would still work per the RedHat Security recommendation (one owner) while also immediately updating the team contacts. Perhaps it can be:
- send your alert to `quarkus-secalert@redhat.com` (note your alert will be immediately forwarded to `secalert@redhat.com` and the Quarkus security team). 
- If you prefer you can send an alert directly to `secalert@redhat.com`, for example, if you need to encrypt the message content with a public key available [here](https://access.redhat.com/security/team/contact).
something like that. If you are concerned it can be too messy then `secalert@redhat.com` only is fine

File: SECURITY.md
Comment (maxandersen) [inline SECURITY.md:45]:
```suggestion
The Red Hat security team is a trusted member of the security community.
```

Comment (emmanuelbernard) [reply SECURITY.md:45]: fixed (forgot to push)

**Comment by stianst** [comment]
@emmanuelbernard this mail boil down to difference in our projects. In most cases people from the Keycloak team has better experience and understanding of security issues within our project than the Red Hat security team. Also, most of the team are interested in these, and should be aware of the issues.

**Comment by gsmet** [comment]
@emmanuelbernard I don't know what's behind it but apparently there's a GH feature to define a security policy here: https://github.com/quarkusio/quarkus/security .

**Comment by maxandersen** [comment]
> @emmanuelbernard I don't know what's behind it but apparently there's a GH feature to define a security policy here: https://github.com/quarkusio/quarkus/security .

that policy creation is what makes SECURITY.md PR :)

**Comment by maxandersen** [comment]
I feel `security@quarkus.io` which initially have `secalert@redhat.com` as  the receiver allowing for in future those in the community that are willing to help out with security can be added.

**Comment by sberyozkin** [comment]
Hi Max,
> I feel security@quarkus.io which initially have secalert@redhat.com as the receiver allowing for in future those in the community that are willing to help out with security can be added is the best approach

It is probably not going to work given that a given vulnerability may have wider implications.

But I do feel that indeed, if someone from the community chooses to report it then both `secalert@redhat.com` and a few Quarkus team members should become aware of it immediately and `security@quarkus.io` gives this option. 
Users can always send to `secalert@redhat.com` directly anyway if they prefer - it is a publicly visible contact.

**Comment by maxandersen** [comment]
> It is probably not going to work given that a given vulnerability may have wider implications.

Not following what your concern is here ? 
It is not like the list of subscribers will be open-ended. It should be people that are trusted.

**Comment by sberyozkin** [comment]
@maxandersen I'm totally behind the idea of making it as open as possible, but if the secalert comes back and says this product version is actually affected then we break their requirements of only having the people in the know updated, this is my understanding at least.

But at the very least we should have a policy IMHO to have non-embargoed CVEs (once vetted by the secalert) reported publicly

**Comment by emmanuelbernard** [comment]
Clarifying a few things

1. what is security@quarkus.io? An alias? Mailman? an alias to a google groups (not even sure that be made to work)
2. by definition, the people that will be notified of a security vulnerability will be a limited set. It won't be open.

As soon as the vulnerability is analyzed and fix put upstream we can open a proper issue and disclose.

**Comment by sberyozkin** [comment]
> what is security@quarkus.io

May be `quarkus-security@redhat.com` or `quarkus-secalert@redhat.com` ?

**Comment by sberyozkin** [comment]
If the requests go through the `quarkus-security@redhat.com` or `quarkus-secalert@redhat.com` then it will be just easier for us to build a picture of the currently opened requests faster.
It will also be great to have `https://quarkus.io/security/cve` where the non-embargoed CVEs are published once they have been fixed.

**Comment by emmanuelbernard** [comment]
OK all, I've updated the security policy proposal. I'm testing the system but in the mean time let's finish the proposal.

File: SECURITY.md
Comment (sberyozkin) [inline SECURITY.md:40]: How it can be demonstrated though without a given individual being on this list from the start ? And would such an individual see the issues which someone else would like to be embargoed or which are affecting the product ? 

It feels that 
```
The subscribers are only trusted individuals from the Quarkus community who will handle the resolution.
```
From the 1st section is sufficient - it keeps an option open for yourself to invite from the wider community without going into the specifics.

**Comment by sberyozkin** [comment]
@emmanuelbernard Thanks for requesting a review, it looks very good IMHO, I've left 1 comment only

File: SECURITY.md
Comment (emmanuelbernard) [reply SECURITY.md:40]: The idea was to express that we were open but that the people should have creds.
In practice, some experience with handling security issues is what we would like (from other contexts). People part of that circle will be automatically embargoed and some embargoed lift would need to be coordinated with all parties involved. So that's not a trivial element but something security teams know how to do (see Spectre, Meltdown as example of industry wide embargoe and coordination.

Comment (emmanuelbernard) [reply SECURITY.md:40]: I am wonderI'm adding " (e.g. experience with embargoe process)"

**Review by maxandersen — APPROVED** [review]
+1 looks good to me.

File: SECURITY.md
Comment (sberyozkin) [reply SECURITY.md]: @emmanuelbernard makes sense, thanks

**Comment by emmanuelbernard** [comment]
Something is wrong with coming out of Draft and the Review request gatewkeeper. Will push myself

**Pull request closed without merging**
